### DISCUSSION TITLE
Anomalous console prints during the solve on older HPC

### DISCUSSION THREAD

**Discussion by mangerij** [idaholab/moose#18908] [Q&A General] [answered]
Hi,

I noticed after a recent recompile (fresh MOOSE/libMesh/Petsc stack) on an older HPC system that I have console prints telling me the following:

Adapting Mesh
    Adapting Mesh.....                                                                   7.03 s
  Still Adapting Mesh................................................

  Still Adapting Mesh..                                                                        60.40 s

or

 Finished Computing Residual                                                         6.99 s
    Computing Residual
      Updating Displaced Mesh.....                                                       6.81 s
    Finished Computing Residual                                                          7.00 s

for what it appears to be every _linear_ iteration. 

The problem is a bit large, with some aggressive AMR active (giving something like 5000 dof/proc when max h_level is reached). We are not using displaced mesh also so losing 6.81 seconds for every linear iteration is a bit much for me. 

Has anyone seen this before?

**Comment by GiudGiud** [comment]
Hello

Can you share your input file? We'd have to look to find what is using the displaced mesh. 

Guillaume

**Reply by mangerij** [reply]
We are only using the tensor_mechanics Action that sets up the stress-divergence Kernels and the GlobalStrain system to ensure strain periodicity.

I should note that I have noticed the "update displaced mesh issue" for quite some time. I had a thread on the old google group about this.

I have a much simpler input file than the problem above that does not use AMR nor print "Computing Residual" but **does** print "updating displaced mesh" when the dof/proc gets large enough. Maybe that is the best place to start. Can I send that off-list to you? thanks I will also have to give you access to Ferret.

**Reply by GiudGiud** [reply]
You can send me that file, either on the email listed on github or on my INL email (swap out gmail for inl.gov)

**Reply by friedmud** [reply]
I know that the tensor_mechanics action can definitely turn on the displaced mesh... it uses it any time you have a large strain formulation (and possibly in other scenarios too).

Is the updating of the displaced mesh the main issue here?  Or are you wanting to turrn off the printing?  You can turn off the printing by using `perf_graph_live = false` in your `Outputs` block... or you can turn the time up that it waits before printing by setting `perf_graph_live_time_limit = #` where `#` is however many seconds you want it to wait before printing anything.

This is a new system for printing progress that we're still refining.  Feel free to provide feedback on it.

**Reply by mangerij** [reply]
@GiudGiud I granted your github acct Ferret access and sent the input file to your gmail. Thanks!

EDIT @friedmud indeed I put `perf_graph_live = false` in the outputs block and I am no longer seeing the update displaced mesh (along with computing residual, jacobian etc) _prints_. However, the problem is damn slow now -  and I worry there is some needless checks related to Update displaced mesh that is causing this.

**Reply by friedmud** [reply]
Yeah - if your issue is that PerfGraphLive _exposed_ something happening that shouldn't be... then that's a different issue.

**Reply by dschwen** [reply]
The TM Action is not turning displaced mesh on, but as soon as you put `displacements` into `[GlobalParams]` it is also picked up by the `[Mesh]` block and will trigger the creation of the displaced mesh.

**Comment by hugary1995** [comment]
I've seen similar behavior when I loaded an MPI different from the one used for compilation.

**Reply by mangerij** [reply]
I do get the warning

MPI startup(): Warning: I_MPI_PMI_LIBRARY will be ignored since the hydra process manager was found

at the start up, but I am relatively certain we are using the same MPI during compilation (no change to the loaded modules, etc)

**Comment by GiudGiud** [comment, marked as answer]
Solution is to set use_displaced_mesh in the [Mesh] block to false when providing displacements. 
We ll document that, raised an issue https://github.com/idaholab/moose/issues/19049

**Reply by mangerij** [reply]
Thanks @GiudGiud and @dschwen
